stats/B161/feedback-analysis.ipynb

```diff
@@ -0,0 +1,1130 @@
+      "text/plain": [
+       "<matplotlib.figure.Figure at 0x7f309b3fada0>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "plot = marks.stack().groupby(level=1).mean().T.plot.bar(\n",
+    "    figsize=(7, 3),\n",
+    "    title='Average ratings of individual lessons'\n",
+    ")\n",
+    "plot.legend(loc='center left', bbox_to_anchor=(1, 0.5))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/html": [
+       "<div>\n",
+       "<table border=\"1\" class=\"dataframe\">\n",
+       "  <thead>\n",
+       "    <tr>\n",
+       "      <th>category</th>\n",
+       "      <th colspan=\"10\" halign=\"left\">like</th>\n",
+       "      <th>...</th>\n",
+       "      <th colspan=\"10\" halign=\"left\">learn</th>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>lesson</th>\n",
+       "      <th>requests-click</th>\n",
+       "      <th>flask</th>\n",
+       "      <th>moduly</th>\n",
+       "      <th>testovani</th>\n",
+       "      <th>dokumentace</th>\n",
+       "      <th>pandas</th>\n",
+       "      <th>numpy</th>\n",
+       "      <th>cython</th>\n",
+       "      <th>pyqt</th>\n",
+       "      <th>async</th>\n",
+       "      <th>...</th>\n",
+       "      <th>testovani</th>\n",
+       "      <th>dokumentace</th>\n",
+       "      <th>pandas</th>\n",
+       "      <th>numpy</th>\n",
+       "      <th>cython</th>\n",
+       "      <th>pyqt</th>\n",
+       "      <th>async</th>\n",
+       "      <th>magie</th>\n",
+       "      <th>micropython</th>\n",
+       "      <th>semestralka</th>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>value</th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>5.0</th>\n",
+       "      <td>10.0</td>\n",
+       "  
```

Comment:
Not sure if that would help.

I did plot the ratings by individual students as a line graph (interpolated so they don't overlap in the graph):

```python
data = marks.T.swaplevel().loc['topic']#.iloc[4:-1]
cols = data.index.copy()
data = data.reset_index(drop=True)
data.index *= 10
for i in range(len(data) * 10 - 9):
    if i % 10:
        data.loc[i] = float('nan')
data = data.interpolate('akima').sort_index()
plot = data.plot(rot=90, legend=False, figsize=(15, 10))
plot.set_xticks([i*10 for i, c in enumerate(cols)])
plot.set_xticklabels(cols);
```

and saw that all the lines are crazy, and are no straight lines at the bottom (which would be students that are permanently unsatisfied):

![index](https://cloud.githubusercontent.com/assets/302922/25392136/245b5806-29d8-11e7-90e1-e4664a589ffe.png)

But I guess such students wouldn't fill out the survey at all, so it's pretty pointless to look at the data here.